lib/string-checker.js

```diff
@@ -136,8 +150,87 @@ StringChecker.prototype = {
 
             this._errorsFound += errors.getErrorCount();
         }
+    },
 
-        return errors;
+    /**
+     * Adds parse error to the error list.
+     *
+     * @param {Errors} errors
+     * @param {Error} parseError
+     * @private
+     */
+    _addParseError: function(errors, parseError) {
+        errors.setCurrentRule('parseError');
+        errors.add(parseError.description, parseError.lineNumber, parseError.column);
+    },
+
+    /**
+     * Creates configured JsFile instance.
+     *
+     * @param {String} filename
+     * @param {String} source
+     * @param {Object} sourceTree
+     * @private
+     */
+    _createJsFileInstance: function(filename, source, sourceTree) {
+        return new JsFile(filename, source, sourceTree, {
+            es3: this._configuration.isES3Enabled(),
+            es6: this._configuration.isESNextEnabled()
+        });
+    },
+
+    /**
+     * Checks file provided with a string.
+     * @param {String} source
+     * @param {String} [filename]
+     * @returns {{output: String, errors: Errors}}
+     */
+    fixString: function(source, filename) {
+        filename = filename || 'input';
+
+        var sourceTree;
+        var parseError;
+
+        try {
+            sourceTree = JsFile.parse(source, this._esprima, this._configuration.getEsprimaOptions());
+        } catch (e) {
+            parseError = e;
+        }
+
+        if (parseError) {
+            var parseErrors = new Errors(this._createJsFileInstance(filename, source, sourceTree), this._verbose);
+            this._addParseError(parseErrors, parseError);
+            return {output: source, errors: parseErrors};
+        } else {
+            var attempt = 0;
+            var errors;
+            var file;
+
+            do {
+                file = this._createJsFileInstance(filename, source, sourceTree);
+                errors = new Errors(file, this._verbose);
+                this._checkJsFile(file, errors);
+                var hasFixes = false;
+                var errorList = errors.getErrorList();
+                for (var i = 0; i < errorList.length; i++) {
+                    if (errorList[i].fixed) {
+                        hasFixes = true;
+                        break;
+                    }
+                }
+
+                if (!hasFixes) {
+                    break;
+                }
+
+                source = file.render();
+                sourceTree = JsFile.parse(source, this._esprima, this._configuration.getEsprimaOptions());
+
+                attempt++;
+            } while (attempt < MAX_FIX_ATTEMPTS);
```

Comment:
> Keep looping as long as fixes > 0 and violations < violationsOfPreviousRun.

This might not work. For instance one rule does 1 fix and then there are 2 or more fixes needs to be done also.